src/sys/Net/NetServices.cs

```diff
@@ -40,6 +40,7 @@ public class NetServices
         private const int LOCAL_ADDRESS_CACHE_LIFETIME_SECONDS = 300;   // The amount of time to leave the result of a local IP address determination in the cache.
 
         private static ILogger logger = Log.Logger;
+        private static NetworkInterface[] adapters = NetworkInterface.GetAllNetworkInterfaces(); // Assume this will not change while the program runs
 

```

Comment:
That behaviour was used in the past but was deliberately removed. This library can potentially run on mobile devices where it is likely the interfaces could change during an application lifetime.

Reply:
The adapters do not change - except when a physical device is plugged in. The array provides a list of known physycal devices, which I do not see changed during the lifetime of an app. Note, this is not related to the fact that an adapter is up or down. The array only provides a list of physical devices. I tested this on my Android and iOS devices by enabling/disabling WiFi and cellular and restarting the app. The array stays identical.